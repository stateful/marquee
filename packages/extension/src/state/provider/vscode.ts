import type { ExtensionContext } from 'vscode'
import type { StateProvider } from '../types'

export class VSCodeState implements StateProvider {
  constructor (public context: ExtensionContext) {}

  getState (key: string) {
    return this.context.globalState.get(key) as never
  }

  async setState (key: string, value: any) {
    await this.context.globalState.update(key, value)
  }
}
