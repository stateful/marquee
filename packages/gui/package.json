{
  "name": "@vscode-marquee/gui",
  "description": "GUI of Marquee",
  "version": "0.1.0",
  "private": true,
  "devDependencies": {},
  "dependencies": {
    "@emotion/core": "^11.0.0",
    "@emotion/styled": "^11.6.0",
    "@fortawesome/free-solid-svg-icons": "^6.0.0",
    "@fortawesome/react-fontawesome": "^0.1.17",
    "@material-ui/core": "^4.12.3",
    "@material-ui/icons": "^4.11.2",
    "@material-ui/lab": "^4.0.0-alpha.60",
    "@sentry/integrations": "^6.17.6",
    "@sentry/react": "^6.17.6",
    "@sentry/tracing": "^6.17.6",
    "@types/emoji-mart": "^3.0.9",
    "@types/react-dom": "^17.0.13",
    "@types/react-grid-layout": "^1.3.1",
    "@types/react-portal": "^4.0.4",
    "@types/vscode": "^1.64.0",
    "@vscode-marquee/dialog": "^0.1.0",
    "@vscode-marquee/utils": "^0.1.0",
    "@vscode-marquee/widget": "^0.1.0",
    "@vscode-marquee/widget-github": "^0.1.0",
    "@vscode-marquee/widget-news": "^0.1.0",
    "@vscode-marquee/widget-notes": "^0.1.0",
    "@vscode-marquee/widget-projects": "^0.1.0",
    "@vscode-marquee/widget-snippets": "^0.1.0",
    "@vscode-marquee/widget-todo": "^0.1.0",
    "@vscode-marquee/widget-weather": "^0.1.0",
    "@vscode-marquee/widget-welcome": "^0.1.0",
    "copy-to-clipboard": "^3.3.1",
    "email-validator": "^2.0.4",
    "emoji-mart": "^3.0.1",
    "html-escaper": "^3.0.3",
    "react": "^17.0.2",
    "react-animated-slider": "^2.0.0",
    "react-debounce-input": "^3.2.5",
    "react-dom": "^17.0.2",
    "react-editext": "^4.2.1",
    "react-fast-compare": "^3.2.0",
    "react-grid-layout": "^1.3.3",
    "react-live-clock": "^5.6.1",
    "react-movable": "^3.0.2",
    "react-portal": "^4.2.1",
    "react-rnd": "^10.3.5",
    "react-split-pane": "^0.1.92",
    "react-splitter-layout": "^4.0.0",
    "react-truncate": "^2.4.0",
    "react-virtualized": "^9.22.3",
    "tangle": "^1.0.5",
    "typeface-lato": "^1.1.13",
    "use-debounce": "^7.0.1",
    "weather-icons-react": "^1.2.0"
  }
}
