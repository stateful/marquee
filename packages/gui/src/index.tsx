import React, { useEffect } from 'react'
import ReactDOM from 'react-dom'

import CssBaseline from '@mui/material/CssBaseline'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'

import { theme, GlobalProvider } from '@vscode-marquee/utils'
import type { MarqueeWindow } from '@vscode-marquee/utils'

import Sentry from './sentry'
import Container from './Container'

import { ModeProvider } from './contexts/ModeContext'

import './css/index.css'


declare const window: MarqueeWindow

window.vscode = window.acquireVsCodeApi() as MarqueeWindow['vscode']
Sentry.init()

export const Providers = ({ children }: any) => {
  return (
    <ModeProvider>
      <GlobalProvider>
        {children}
      </GlobalProvider>
    </ModeProvider>
  )
}

export const App = () => {
  /**
   * tell extension backend that application is readyy
   */
  useEffect(() => { window.vscode.postMessage({ ready: true }) }, [])

  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <Providers>
          <Container />
        </Providers>
      </ThemeProvider>
    </StyledEngineProvider>
  )
}

if (process.env.NODE_ENV !== 'test') {
  ReactDOM.render(<App />, document.getElementById('root'))
}
