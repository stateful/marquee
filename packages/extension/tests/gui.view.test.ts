import vscode from 'vscode';
import { MarqueeGui } from '../src/gui.view';

const stateMgr = {
  widgetExtensions: [
    { exports: { marquee: { disposable: { on: jest.fn() } } } },
    { exports: { marquee: { disposable: { on: jest.fn() } } } }
  ]
};

test('constructor', () => {
  new MarqueeGui('context' as any, stateMgr as any);
  expect(stateMgr.widgetExtensions[0].exports.marquee.disposable.on).toBeCalledTimes(2);
  expect(stateMgr.widgetExtensions[0].exports.marquee.disposable.on)
    .toBeCalledWith('gui.open', expect.any(Function));
  expect(stateMgr.widgetExtensions[0].exports.marquee.disposable.on)
    .toBeCalledWith('gui.close', expect.any(Function));
});

test('isActive', () => {
  const gui = new MarqueeGui('context' as any, stateMgr as any);
  gui['guiActive'] = 'foobar' as any;
  expect(gui.isActive()).toBe('foobar');
});

test('close', () => {
  const gui = new MarqueeGui('context' as any, stateMgr as any);
  gui.close();
  gui['panel'] = { dispose: jest.fn() } as any;
  gui.close();
  expect(gui['panel']!.dispose).toBeCalledTimes(1);
});

test('broadcast', () => {
  const gui = new MarqueeGui('context' as any, stateMgr as any);
  expect(gui.broadcast('openSettings', 'foobar')).toBe(false);

  gui['client'] = { emit: jest.fn() } as any;
  expect(gui.broadcast('openSettings', 'foobar')).toBe(undefined);
  expect(gui['client']!.emit).toBeCalledWith('openSettings', 'foobar');
});

test('_executeCommand', () => {
  const gui = new MarqueeGui('context' as any, stateMgr as any);
  gui['_executeCommand']({ command: 'vscode.openFolder', args: ['foo', 'bar'], options: 'foobar' as any});
  expect(vscode.commands.executeCommand).toBeCalledWith('vscode.openFolder', 'parsedUri-foo', 'foobar');

  gui['_executeCommand']({ command: 'barfoo', args: ['foo', 'bar'], options: 'foobar' as any});
  expect(vscode.commands.executeCommand).toBeCalledWith('barfoo', 'foo', 'foobar');

  gui['_executeCommand']({ command: 'barfoo', options: 'foobar'} as any);
  expect(vscode.commands.executeCommand).toBeCalledWith('barfoo', undefined, 'foobar');
});

test('_handleNotifications', () => {
  const gui = new MarqueeGui('context' as any, stateMgr as any);
  gui['_handleNotifications']({ type: 'error', message: 'foobar' });
  expect(vscode.window.showErrorMessage).toBeCalledWith('foobar');

  gui['_handleNotifications']({ type: 'warning', message: 'foobar' });
  expect(vscode.window.showWarningMessage).toBeCalledWith('foobar');

  gui['_handleNotifications']({ type: 'anything', message: 'foobar' });
  expect(vscode.window.showInformationMessage).toBeCalledWith('foobar');
});
