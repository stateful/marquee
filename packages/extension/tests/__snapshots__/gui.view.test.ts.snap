// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open webview 1`] = `
Object {
  "dark": "/some/uri/assets/marquee-tab.svg",
  "light": "/some/uri/assets/marquee-tab.svg",
}
`;

exports[`open webview 2`] = `
Array [
  Array [
    "<html></html>",
    Object {
      "aws": Object {
        "id": "foobar",
      },
      "baseAppUri": "some url",
      "baseUrl": "http://BACKEND_BASE_URL.com",
      "childSrc": Array [
        "https://www.youtube.com",
        "https://player.vimeo.com",
        "https://fast.wistia.net",
      ],
      "colorScheme": "transparent",
      "connectSrc": Array [
        "undefined://undefined",
        "undefined://undefined",
        "https://*.ingest.sentry.io",
      ],
      "cspSource": "csp-source",
      "fontSize": 1,
      "fwdGeoUrl": "http://BACKEND_FWDGEO_URL.com",
      "geoUrl": "http://BACKEND_GEO_URL.com",
      "nonce": "MzlkMmY4NTgtMzUyMi00MDUzLWJlYjgtYjc1YTFkZWZkMWQy",
      "props": Object {
        "some": "props",
      },
      "widgetScripts": Array [],
      "widgetStateConfigurations": "eyJ1bmRlZmluZWQiOnsiY29uZmlndXJhdGlvbiI6e30sInN0YXRlIjp7fX19",
    },
  ],
]
`;
